include/secp256k1.h

```diff
@@ -580,16 +579,17 @@ SECP256K1_API int secp256k1_ecdsa_sign(
     const void *ndata
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
 
-/** Same as secp256k1_ecdsa_sign, but nonce_tweak32 is added to the nonce generated by noncefp.
+/** Same as secp256k1_ecdsa_sign, but s2c_data32 is committed to by adding `hash(R1, s2c_data32)` to
+    the nonce generated by noncefp, with `ndata=hash(s2c_data32, ndata)`.
  */
-SECP256K1_API int secp256k1_ecdsa_sign_nonce_tweak_add(
+SECP256K1_API int secp256k1_ecdsa_sign_to_contract(
     const secp256k1_context* ctx,
     secp256k1_ecdsa_signature *sig,
     const unsigned char *msg32,
     const unsigned char *seckey,
     secp256k1_nonce_function noncefp,
     const void *ndata,
-    const unsigned char* nonce_tweak32
+    const unsigned char* s2c_data32
```

Comment:
nit: we should standardize on the position (in my PR it's before the noncefp, but I don't really care which way it is).